### PR TITLE
Add initial docs for the /pyroscope/render API endpoint

### PR DESCRIPTION
**Pull request by aleks-p**

### CODE REVIEW

File: docs/sources/configure-server/about-server-api.md [hunk 1]
```diff
@@ -21,16 +21,16 @@ There is one primary endpoint: POST /ingest. It accepts profile data in the requ
...
-| `spyName`          | name of the spy used                    | optional                        |
-| `units`            | name of the profiling data unit         | optional (default is `samples`  |
-| `aggregrationType` | type of aggregration to merge profiles  | optional (default is `sum`)     |
+| Name               | Description                             | Notes                          |
+|:-------------------|:----------------------------------------|:-------------------------------|
+| `name`             | application name                        | required                       |
+| `from`             | UNIX time of when the profiling started | required                       |
+| `until`            | UNIX time of when the profiling stopped | required                       |
+| `format`           | format of the profiling data            | optional (default is `folded`) |
+| `sampleRate`       | sample rate used in Hz                  | optional (default is 100 Hz)   |
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:30]: Should 100 be `100`?

File: docs/sources/configure-server/about-server-api.md [hunk 2]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
 {{< /code >}}
 
+## Querying Profile Data
+
+There is one primary endpoint for querying profile data: `GET /pyroscope/render`.
+
+The search input is provided via query parameters.
+The output is typically a JSON object containing one or more time series and a flamegraph.
+
+### Query Parameters
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:223]: Changing to sentence case. 
```suggestion
### Query parameters
```

File: docs/sources/configure-server/about-server-api.md [hunk 3]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
 
 {{< /code >}}
 
+## Querying Profile Data
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:216]:
```suggestion
## Querying profile data
```

File: docs/sources/configure-server/about-server-api.md [hunk 4]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+
+Here is a specific example:
+
+`process_cpu:cpu:nanoseconds:cpu:nanoseconds{service_name="my_application_name"}`
+
+In a Kubernetes environment, a query could also look like:
+
+`process_cpu:cpu:nanoseconds:cpu:nanoseconds{namespace="dev", container="my_application_name"}`
+
+Note: Take a look at [profiling types]({{< relref "../ingest-and-analyze-profile-data/profiling-types" >}}) for more information.
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:251]: We're using an admonition shortcode for notes, cautions, tips, and warnings. For more information, see the [Shortcodes](https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition) documentation. 
```suggestion
{{% admonition type="note" %}}
Take a look at [profiling types]({{< relref "../ingest-and-analyze-profile-data/profiling-types" >}}) for more information.
{{% /admonition %}}
```

File: docs/sources/configure-server/about-server-api.md [hunk 5]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+
+#### `from` and `until`
+
+The `from` and `until` parameters determine the start and end of the time period that we want to query for.
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:255]:
```suggestion
The `from` and `until` parameters determine the start and end of the time period for the query.
```

File: docs/sources/configure-server/about-server-api.md [hunk 6]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+They can be provided in absolute and relative form.
+
+**Absolute time**
+
+When passing absolute values we have a few options:
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:260]:
```suggestion
This table details the options for passing absolute values.
```

File: docs/sources/configure-server/about-server-api.md [hunk 7]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+
+The format can either be:
+- `json`, in which case the response will contain a JSON object
+- `dot`, in which case the response will be text containing a DOT representation of the profile
+
+See the [Query Output](#query-output) section for more information on the response structure.
+
+#### `maxNodes`
+
+The `maxNodes` parameter truncates the number of elements in the profile response, to allow tools (e.g., a frontend) to render large profiles efficiently.
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:302]:
```suggestion
The `maxNodes` parameter truncates the number of elements in the profile response, to allow tools (for example, a frontend) to render large profiles efficiently. 
```

File: docs/sources/configure-server/about-server-api.md [hunk 8]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+#### `metadata`
+
+The `metadata` field contains additional information that is helpful to interpret the `flamebearer` data such as the unit (nanoseconds, bytes), sample rate and more.
+
+#### `timeline`
+
+The `timeline` field represents the time series for the profile.
+Pyroscope pre-computes the step interval (resolution) of the timeline using the query interval (`from` and `until`). The minimum step interval is 10 seconds.
+
+The raw profile sample data is down-sampled to the step interval (resolution) using an aggregation function (currently only `sum` is supported).
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:342]:
```suggestion
The raw profile sample data is down-sampled to the step interval (resolution) using an aggregation function. Currently, only `sum` is supported.
```

File: docs/sources/configure-server/about-server-api.md [hunk 9]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+      400
+    ],
+    "durationDelta": 10
+  }
+}
+```
+
+### Alternative Query Output
+
+When the `format` query parameter is `dot`, the endpoint will respond with a [DOT format](https://en.wikipedia.org/wiki/DOT_(graph_description_language)) data representing the queried profile.
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:362]:
```suggestion
When the `format` query parameter is `dot`, the endpoint responds with a [DOT format](https://en.wikipedia.org/wiki/DOT_(graph_description_language)) data representing the queried profile.
```

File: docs/sources/configure-server/about-server-api.md [hunk 10]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+This can be used to create an alternative visualization of the profile.
+
+### Example Queries
+
+Here is an example where we query a local Pyroscope server for a CPU profile from the `pyroscope` service for the last hour.
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:367]:
```suggestion
### Example queries

This example queries a local Pyroscope server for a CPU profile from the `pyroscope` service for the last hour.
```

Comment (knylander-grafana) [reply docs/sources/configure-server/about-server-api.md:367]: Great job providing multiple examples.

File: docs/sources/configure-server/about-server-api.md [hunk 11]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+```
+
+See [this Python script](https://github.com/grafana/pyroscope/tree/main/examples/api/query.py) for a complete example.
+
+#### `groups`
+
+The `groups` field is only populated when grouping is requested by the `groupBy` query parameter.
+When this is the case, the `groups` field will have an entry for every label value found for the query.
+
+Example where we group by a cluster:
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:403]:
```suggestion
When this is the case, the `groups` field has an entry for every label value found for the query.

This example groups by a cluster:
```

File: docs/sources/configure-server/about-server-api.md [hunk 12]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+  "groups": {
+    "eu-west-2": { "startTime": 1577836800, "samples": [ 200, 300, 500 ] },
+    "us-east-1": { "startTime": 1577836800, "samples": [ 100, 200, 400 ] }
+  }
+}
+```
+
 ## Profile CLI
 
 The `profilecli` tool can also be used to interact with the Pyroscope server API. It supports operations such as ingesting profiles, querying for existing profiles and more. Refer to the [Profile CLI]({{< relref "../ingest-and-analyze-profile-data/profile-cli" >}}) page for more information.
```
Comment (knylander-grafana) [inline docs/sources/configure-server/about-server-api.md:416]:
```suggestion
The `profilecli` tool can also be used to interact with the Pyroscope server API. 
The tool supports operations such as ingesting profiles, querying for existing profiles, and more.
Refer to the [Profile CLI]({{< relref "../ingest-and-analyze-profile-data/profile-cli" >}}) page for more information.
```

**Review by knylander-grafana — APPROVED** [review]
Approving with a few minor suggestions. Thank you for updating and adding doc!

File: docs/sources/configure-server/about-server-api.md [hunk 13]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+| `query`        | contains the profile type and label selectors                                         | required                                             |
+| `from`         | UNIX time for the start of the search window                                          | required                                             |
+| `until`        | UNIX time for the end of the search window                                            | optional (default is `now`)                          |
+| `format`       | format of the profiling data                                                          | optional (default is `json`)                         |
+| `maxNodes`     | the maximum number of nodes the resulting flamegraph will contain                     | optional (default is `max_flamegraph_nodes_default`) |
+| `groupBy`      | one or more label names to group the time series by (doesn't apply to the flamegraph) | optional (default is no grouping)                    |
+
+#### `query`
+
+The `query` parameter is the only required search input. It carries the profile type and any labels we want to use to narrow down the output.
```
Comment (Rperry2174) [inline docs/sources/configure-server/about-server-api.md:238]:
```suggestion
The `query` parameter is the only required search input. It carries the profile type and any labels we want to use to narrow down the output. You can check here for the list of [valid profile types](https://github.com/grafana/pyroscope/blob/main/public/app/constants/profile-metrics.json)
```

File: docs/sources/configure-server/about-server-api.md [hunk 14]
```diff
@@ -213,6 +213,204 @@ curl -X POST \
...
+| 3 hours ago    | `now-3h`             |
+| 30 minutes ago | `now-30m`            |
+| 2 days ago     | `now-2d`             |
+| 1 week ago     | `now-7d` or `now-1w` |
+
+Note that a single offset has to be provided, values such as `now-3h30m` will not work.
+
+**Validation**
+
+The `from` and `until` parameters are subject to these validation rules:
```
Comment (Rperry2174) [inline docs/sources/configure-server/about-server-api.md:285]:
```suggestion
The `from` and `until` parameters are subject to these validation rules related to `max_query_lookback` and `max_query_length` server properties. You can find more details on these properties in the server docs. 
```

Not sure if we actually mention these parameters in our server docs, but if so we should link to them

Comment (aleks-p) [reply docs/sources/configure-server/about-server-api.md:285]: Good idea! We have them in the "limits" section, will add a link.

**Pull request merged**